pkg/pipelines/environments/environments_test.go

```diff
@@ -135,36 +143,64 @@ func filesFromResources(r res.Resources) []string {
 }
 
 func buildManifest(withCICD bool) *config.Manifest {
-	cfg := &config.Manifest{
-		Environments: []*config.Environment{
-			{
-				Name: "test-dev",
-				Apps: []*config.Application{
-					{
-						Name: "my-app-1",
-						ServiceRefs: []string{
-							"service-http",
-							"service-metrics",
+	if withCICD {
+		return &config.Manifest{
+			Config: &config.Config{
+				CICD: &config.Cicd{
+					Namespace: "cicd",
+				},
+			},
+			Environments: []*config.Environment{
+				{
+					Name: "test-dev",
+					Apps: []*config.Application{
+						{
+							Name: "my-app-1",
+							ServiceRefs: []string{
+								"service-http",
+								"service-metrics",
+							},
+						},
+					},
+					Services: []*config.Service{
+						{
+							Name:      "service-http",
+							SourceURL: "https://github.com/myproject/myservice.git",
+						},
+						{
+							Name: "service-metrics",
 						},
 					},
 				},
-				Services: []*config.Service{
-					{
-						Name:      "service-http",
-						SourceURL: "https://github.com/myproject/myservice.git",
+			},
+		}
+	}
+	if !withCICD {
+		return &config.Manifest{
+			Environments: []*config.Environment{
+				{
+					Name: "test-dev",
+					Apps: []*config.Application{
+						{
+							Name: "my-app-1",
+							ServiceRefs: []string{
+								"service-http",
+								"service-metrics",
+							},
+						},
 					},
-					{
-						Name: "service-metrics",
+					Services: []*config.Service{
+						{
+							Name:      "service-http",
+							SourceURL: "https://github.com/myproject/myservice.git",
+						},
+						{
+							Name: "service-metrics",
+						},
 					},
 				},
 			},
-		},
-	}
-	if withCICD == true {
-		cfg.Environments = append(cfg.Environments, &config.Environment{
-			Name:   "cicd",
-			IsCICD: true,
-		})
+		}
 	}
-	return cfg
+	return nil
```

Comment:
The caller downstream does not handle nil.   This will cause the caller to crash.